arangod/Replication2/Methods.cpp

```diff
@@ -591,6 +593,70 @@ struct ReplicatedLogMethodsCoordinator final
   network::ConnectionPool* pool;
 };
 
+struct ReplicatedStateDBServerMethods
+    : std::enable_shared_from_this<ReplicatedStateDBServerMethods>,
+      ReplicatedStateMethods {
+  explicit ReplicatedStateDBServerMethods(TRI_vocbase_t& vocbase)
+      : vocbase(vocbase) {}
+
+  auto createReplicatedState(replicated_state::agency::Target const& spec) const
+      -> futures::Future<Result> override {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+  }
+
+  auto deleteReplicatedLog(LogId id) const -> futures::Future<Result> override {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+  }
+
+  auto getLocalStatus(LogId id) const
+      -> futures::Future<replicated_state::StateStatus> override {
+    auto state = vocbase.getReplicatedStateById(id);
+    if (auto status = state->getStatus(); status.has_value()) {
+      return std::move(*status);
+    }
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_HTTP_SERVICE_UNAVAILABLE);
+  }
+
+  TRI_vocbase_t& vocbase;
+};
+
+struct ReplicatedStateCoordinatorMethods
+    : std::enable_shared_from_this<ReplicatedStateCoordinatorMethods>,
+      ReplicatedStateMethods {
+  explicit ReplicatedStateCoordinatorMethods(TRI_vocbase_t& vocbase)
+      : vocbase(vocbase),
+        clusterInfo(
+            vocbase.server().getFeature<ClusterFeature>().clusterInfo()) {}
+
+  auto createReplicatedState(replicated_state::agency::Target const& spec) const
+      -> futures::Future<Result> override {
+    return replication2::agency::methods::createReplicatedState(vocbase.name(),
+                                                                spec)
+        .thenValue([self = shared_from_this()](
+                       ResultT<uint64_t>&& res) -> futures::Future<Result> {
+          if (res.fail()) {
+            return futures::Future<Result>{std::in_place, res.result()};
+          }
+
+          return self->clusterInfo.waitForPlan(res.get());
+        });
+  }
+
+  auto deleteReplicatedLog(LogId id) const -> futures::Future<Result> override {
```

Comment:
This is not important for the current functionality, but just for my understanding, I assume the intention will be to delete the replicated state.